IntelliTect.TestTools.Data/DatabaseFixture.cs

```diff
@@ -0,0 +1,126 @@
+
+            return ServiceProvider
+                .GetService<ILoggerFactory>();
+        }
+
+        private TDbContext CreateNewContext()
+        {
+            var constructorInfo = typeof(TDbContext)
+                .GetConstructors()
+                .Where(x =>
+                {
+                    var parameters = x.GetParameters();
+                    return parameters.Length == 1 && parameters[0].ParameterType == typeof(DbContextOptions);
+                })
+                .SingleOrDefault();
+
+            if (constructorInfo is null)
+            {
+                throw new InvalidOperationException(
+                    $"'{typeof(TDbContext)}' must contain a constructor that has a single parameter " +
+                    $"of type '{typeof(DbContextOptions)}'");
+            }
+
+            bool alreadyCreated = Options.IsValueCreated;
+
+            var db = (TDbContext) constructorInfo.Invoke(new object[]{ Options.Value });
+
+            if (!alreadyCreated)
+            {
+                db.Database.EnsureCreated();
+            }
+
+            return db;
+        }
+
+        /// <summary>
+        /// Creates new instance of TDBContext and executes database operation.
+        /// This avoids issues where reusing the same DbContext can result in cached objects being returned,
+        /// suppressing issues with your LINQ to SQL code.
+        /// At minimum the Arrange/Act/Assert portions should each invoke this method separately, but more invocations
+        /// of this method should be preferred whenever possible.
+        /// </summary>
+        /// <param name="operation">The database operation to be performed</param>
+        public async Task PerformDatabaseOperation(Func<TDbContext, Task> operation)
+        {
+            var db = CreateNewContext();
+            await operation(db);
+        }
+
+        /// <summary>
+        /// If <see cref="InMemoryLogger"/> is configured and DbContext has been accessed at least once, returns a
+        /// Dictionary of all InMemoryLoggers
+        /// </summary>
+        /// <returns>Dictionary with category name, and instance of all InMemoryLoggers</returns>
+        /// <exception cref="InvalidOperationException">InMemoryLogger is not configured, or DbContext has not yet
+        /// been accessed</exception>
+        public ConcurrentDictionary<string,InMemoryLogger> GetInMemoryLoggers()
+        {
+            if (ServiceProvider is null)
+            {
+                throw new InvalidOperationException("ServiceCollection is not yet initialized. " +
+                                                    "Perform some database operation to initialize loggers");
+            }
+
+            if (!(ServiceProvider.GetService<ILoggerProvider>() is InMemoryLoggerProvider loggerProvider))
+            {
+                throw new InvalidOperationException($"{typeof(ILoggerProvider)} of type " +
```

Comment:
Did you mean to use `typeof` to get the full name (in which case I would recommend being explicit and calling the FullName property), or did you mean to use `nameof` here?

Reply:
I think I did mean to use typeof. Whats the difference between letting it call ToString implicitly vs full name property? Is this a convention?

Reply:
One is assumed behavior that (though very unlikely) could change, the other is explicit.